allensdk/brain_observatory/behavior/data_objects/metadata/behavior_ophys_metadata.py

```diff
@@ -0,0 +1,134 @@
+
+
+class BehaviorOphysMetadata(DataObject, InternalReadableInterface,
+                            JsonReadableInterface, NwbReadableInterface,
+                            NwbWritableInterface):
+    def __init__(self, behavior_metadata: BehaviorMetadata,
+                 ophys_metadata: Union[OphysExperimentMetadata,
+                                       MesoscopeExperimentMetadata]):
+        super().__init__(name='behavior_ophys_metadata', value=self)
+
+        self._behavior_metadata = behavior_metadata
+        self._ophys_metadata = ophys_metadata
+
+    @property
+    def behavior_metadata(self) -> BehaviorMetadata:
+        return self._behavior_metadata
+
+    @property
+    def ophys_metadata(self) -> OphysExperimentMetadata:
+        return self._ophys_metadata
+
+    @classmethod
+    def from_internal(cls, ophys_experiment_id: int,
+                      lims_db: PostgresQueryMixin) -> "BehaviorOphysMetadata":
+        behavior_session_id = BehaviorSessionId.from_lims(
+            ophys_experiment_id=ophys_experiment_id, db=lims_db)
+
+        behavior_metadata = BehaviorMetadata.from_internal(
+            behavior_session_id=behavior_session_id, lims_db=lims_db)
+
+        if behavior_metadata.session_type.is_mesoscope():
+            ophys_metadata = MesoscopeExperimentMetadata.from_internal(
+                ophys_experiment_id=ophys_experiment_id, lims_db=lims_db)
+        else:
+            ophys_metadata = OphysExperimentMetadata.from_internal(
+                ophys_experiment_id=ophys_experiment_id, lims_db=lims_db)
+
+        return cls(behavior_metadata=behavior_metadata,
+                   ophys_metadata=ophys_metadata)
+
+    @classmethod
+    def from_json(cls, dict_repr: dict) -> "BehaviorOphysMetadata":
+        behavior_metadata = BehaviorMetadata.from_json(dict_repr=dict_repr)
+
+        if behavior_metadata.session_type.is_mesoscope():
+            ophys_metadata = MesoscopeExperimentMetadata.from_json(
+                dict_repr=dict_repr)
+        else:
+            ophys_metadata = OphysExperimentMetadata.from_json(
+                dict_repr=dict_repr)
+
+        return cls(behavior_metadata=behavior_metadata,
+                   ophys_metadata=ophys_metadata)
+
+    @classmethod
+    def from_nwb(cls, nwbfile: NWBFile) -> "BehaviorOphysMetadata":
+        behavior_metadata = BehaviorMetadata.from_nwb(nwbfile=nwbfile)
+
+        if behavior_metadata.session_type.is_mesoscope():
+            ophys_metadata = MesoscopeExperimentMetadata.from_nwb(
+                nwbfile=nwbfile)
+        else:
+            ophys_metadata = OphysExperimentMetadata.from_nwb(
+                nwbfile=nwbfile)
+
+        return cls(behavior_metadata=behavior_metadata,
+                   ophys_metadata=ophys_metadata)
+
+    def to_nwb(self, nwbfile: NWBFile) -> NWBFile:
+        self._behavior_metadata.subject_metadata.to_nwb(nwbfile=nwbfile)
+        OphysBehaviorMetadata = load_pynwb_extension(
```

Comment:
could this instance have `nwb` in the name somewhere?
it is awfully close to `BehaviorOphysMetadata` and the distinction would be a little less murky with `nwb` somewhere.

Reply:
Sure, just copied this over from the existing implementation